src/engine/IndexScan.cpp

```diff
@@ -116,9 +116,28 @@ VariableToColumnMap IndexScan::computeVariableToColumnMap() const {
   std::ranges::for_each(additionalVariables_, addCol);
   return variableToColumnMap;
 }
+
 // _____________________________________________________________________________
-ProtoResult IndexScan::computeResult([[maybe_unused]] bool requestLaziness) {
+cppcoro::generator<IdTable> IndexScan::scanInChunks() const {
+  auto metadata = getMetadataForScan(*this);
+  if (!metadata.has_value()) {
+    co_return;
+  }
```

Comment:
I assume that such empty generators work correctly 
(but they should, as you followed the "normal" conventions for for-loops)...

Reply:
They do. In this case `begin() == end()`